client/client.go

```diff
@@ -349,11 +349,12 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
+		resp.Body.Close()
 		<-done
-		if err == nil {
-			err = ctx.Err()
-		}
+		// Because the body is closed, the err var from ReadAll goroutine is
```

Comment:
This comment is not entirely correct. And the behavior is also questionable.

For example, the user call cancel first. Then the client finishes read all of the body and send empty into done. Next, you close the body and receives from done. Then you change nil error to cancel error. 

The next result is that canceled a finished request. Also you returned both error and correct response and its body.